packages/store-sync/src/fromEventSource.ts

```diff
@@ -0,0 +1,10 @@
+import { Observable } from "rxjs";
+
+export function fromEventSource<T>(url: string | URL): Observable<MessageEvent<T>> {
+  return new Observable<MessageEvent>((subscriber) => {
+    const eventSource = new EventSource(url);
+    eventSource.onmessage = (ev): void => subscriber.next(ev);
+    eventSource.onerror = (): void => subscriber.error(new Error("Event source failed" + new URL(url).origin));
```

Comment:
```suggestion
    // we immediately close instead of allowing `EventSource` to retry because the logs API doesn't support `Last-Event-ID` yet
    eventSource.onerror = (): void => subscriber.error(new Error("Event source failed" + new URL(url).origin));
```

Reply:
Since `fromEventSource` is technically not specific to the logs API, we should probably add an option to `fromEventSource` to enable/disable retries, so consumers can decide based on the API capabilities. Would leave for a followup

Reply:
#3246